code/modules/projectiles/updated_projectiles/guns/flamer.dm

```diff
@@ -457,52 +462,38 @@
 	var/burnlevel = 10 //Tracks how HOT the fire is. This is basically the heat level of the fire and determines the temperature.
 	var/flame_color = "red"
 
-/obj/flamer_fire/New(loc, fire_lvl, burn_lvl, f_color, fire_spread_amount, fire_stacks = 0, fire_damage = 0)
-	..()
-	if (f_color)
+/obj/flamer_fire/Initialize(loc, fire_lvl, burn_lvl, f_color, fire_stacks = 0, fire_damage = 0)
+	. = ..()
+
+	if(f_color)
 		flame_color = f_color
 
 	icon_state = "[flame_color]_2"
 	if(fire_lvl)
 		firelevel = fire_lvl
 	if(burn_lvl)
 		burnlevel = burn_lvl
-	START_PROCESSING(SSobj, src)
+	if(fire_stacks || fire_damage)
+		apply_initial_damage(fire_stacks, fire_damage)
 
-	if(fire_spread_amount > 0)
-		var/turf/T
-		for(var/dirn in cardinal)
-			T = get_step(loc, dirn)
-			if(istype(T,/turf/open/space))
-				continue
-			var/obj/flamer_fire/F
-			if(locate(F) in T)
-				qdel(F) //No stacking
-			var/new_spread_amt = T.density ? 0 : fire_spread_amount - 1 //walls stop the spread
-			if(new_spread_amt)
-				for(var/obj/O in T)
-					if(!O.CanPass(src, loc))
-						new_spread_amt = 0
-						break
-			spawn(0) //delay so the newer flame don't block the spread of older flames
-				new /obj/flamer_fire(T, fire_lvl, burn_lvl, f_color, new_spread_amt, fire_stacks, fire_damage)
-				var/mob/living/C
-				if(fire_stacks || fire_damage)
-					for(C in T)
-						if(C.fire_immune)
-							continue
-						else
-							C.adjust_fire_stacks(fire_stacks)
-							var/armor_block = C.run_armor_check("chest", "energy")
-							C.apply_damage(fire_damage, BURN, null, armor_block)
-							C.IgniteMob()
-							C.visible_message("<span class='danger'>[C] bursts into flames!</span>","[isxeno(C)?"<span class='xenodanger'>":"<span class='highdanger'>"]You burst into flames!</span>")
+	START_PROCESSING(SSobj, src)
 
 /obj/flamer_fire/Destroy()
 	SetLuminosity(0)
 	STOP_PROCESSING(SSobj, src)
 	return ..()
 
+/obj/flamer_fire/proc/apply_initial_damage(fire_stacks, fire_damage)
+	var/turf/T = get_turf(src)
+
+	for(var/mob/living/C in T)
+		if(C.fire_immune)
```

Comment:
should be handled by flamer_fire_act

Reply:
Except it doesn't, want me to add it ?

Reply:
if it isnt it should be handled by it, xenos do fire immune differently from memory so its important that behaviour is done via override.

even if you make a new proc called like flamer_fire_direct_act or something